### ISSUE TITLE
[Feature Request] 10-Inch Mini Rack

### ISSUE THREAD

**Issue by TeaRex-coder** [jazwa/rackstack#4]
Hey Jazwa!

I saw your post on [r/homelab](https://www.reddit.com/r/homelab/comments/15wfot0/rackstack_an_opensource_3d_printable_mini_rack) and this seems like a super cool project!

I'd be down to print rackstack in a 10-inch mini rack form factor. I tried to do this myself but couldn't figure out how to do so in OpenSCAD :(

The changes that would have to be made are:
- Increase width to 10 inches (currently possible by adding a custom profile in `rackFrame.scad`)
- Use the EIA-310 rack unit spacing
- Accommodate M6 hardware for rack mount-items
- Increase `mainRail.scad` width and depth to accommodate larger M6 hardware

Thanks for your project, and if you're able to do this I'd 100% be down to test a prototype!

**Comment by jazwa** [comment]
This has been requested by a couple of people already. I'll look into when I get the time. (I don't think it would be a lot of work, but I'm probably going to eat those words later.) Thank you for your interest!

**Comment by lloydw** [comment]
Hi there, I just found your project and it looks amazing! I’m interested in printing a 10” version of this too and was wondering if either of you had got any further with the required changes? Thanks!

**Comment by seantbaker** [comment]
I'm new to networking racks and openSCAD, but am I correct that items 2-4 in the OP are only needed if you're trying to make the printed rack compatible with standard 10" mini racks?

In other words, if I was going to stick to the rackstack ecosystem and print my own mounts for switches, routers, etc., would I only need the first item?

> Increase width to 10 inches (~255mm)  (currently possible by adding a custom profile in rackFrame.scad)

Thanks!

**Comment by TeaRex-coder** [comment]
That's right, you would only need to increase the width to 10" from the outing of each rail or ~9.312" from the inside. Then you could use Rackstack's uniform hole spacing and M3 hardware.

**Comment by TomacGG** [comment]
That's the only thing that i have to do, how to get the 10 inch mounting in the front, the squares?

**Comment by cvrt-jh** [comment]
any progress on this?  printed out the whole thing and figured out later it is not really 10' :D

**Comment by lipusal** [comment]
Hello! Adding my +1 to this feature request. I've spent the last several days trying to do this myself, without success (would gladly submit a PR if I could). I did create a custom profile, I'm messing with the variables on [rackFrame.scad](https://github.com/jazwa/rackstack/blob/main/config/rackFrame.scad) but I can't seem to get to the [supposed dimensions of a 10" rack](https://upload.wikimedia.org/wikipedia/commons/8/84/19_inch_vs_10_inch_rack_dimensions.svg), on width at least. I don't have experience with OpenSCAD and deciphering how everything is formed is proving a challenge (wish there was something like a debugger for this 😅). Would love any pointers here, will try the dimensions shared by @TeaRex-coder 

Also, how do you guys verify dimensions? So far my process is:
1. Modify OpenSCAD files
2. Run `rbuild.py` to generate STLs
3. Import into Fusion 360, convert mesh to solid body
4. If necessary, manually align the parts together
5. Use the Inspect tool to measure

Which is very cumbersome. OpenSCAD nightly supposedly has a measurement feature, but I haven't been able to get it to work.

Thanks!
